docs/v2-faq.md

```diff
@@ -1,22 +1,181 @@
+
+Furthermore, when using the default v2 addon blueprint and its Rollup config, make sure that the `keepAssets` plugin includes the file extensions that you want to import in your addon:
+
+
+```js
+// rollup.config.mjs
+addon.keepAssets(['**/*.css', '**/*.svg']),
+```
+
+Lastly, make sure that the only way you refer to the assets is by using the value returned from the import statement, and _not_ assume what the final URL would be. The bundler is free to choose the final URL, as (depending on your config) it will likely apply some fingerprinting (replacing the legacy `broccoli-asset-rev`), or even choose to inline the asset data (when using `type: 'asset'` in the module rule definition).
+
+### How can I "push" static assets into the app?
+
+Another way to provide the consuming app with static assets from your addon is a push-based approach very similar to the `/public` folder of v1 addons. While the pull-based approach above is preferable in general, in cases where you for example do not need to refer to the asset directly, but want to have a predetermined URL, you can choose the push-based approach instead.
+
+This is done by adding some meta data to the addon's `package.json`, specifying a mapping from the addon's file location to the final public URL, as specified in the [v2 addon spec](https://rfcs.emberjs.com/id/0507-embroider-v2-package-format#assets). Let's say an addon wants to provide a favicon file (which browsers will automatically request from the static `/favicon.ico` URL):
+
+```json
+{
+ "ember-addon": {
+    "version": 2,
+    "type": "addon",
+    "main": "addon-main.cjs",
+    "public-assets": {
+      "./src/assets/public/favicon.ico": "/favicon.ico"
+    }
+  }
+}
+```
+ 
+If you have many files you want to expose this way, you can instead add the `addon.publicAssets()` plugin from `@embroider/addon-dev` to your Rollup config to automate the generation of this mapping data.
+
+## Build setup
+
+### Why do v2 addons need a build step?
+
+The [v2 addon blueprint](https://github.com/embroider-build/addon-blueprint) uses [Rollup](https://rollupjs.org/) for assembling and transpiling your code in to a native npm package that can be imported from anywhere.
+
+While having a build step is not strictly required for v2 addons, there are a few reasons we would want it in general nevertheless:
+- use of pre-shipped JS features (decorators, and other in-progress ECMA proposals)
+- use of TypeScript
+- co-located components are not modules-by-default
+
+The important thing to remember here though is that this build step is very different from the build integration that v1 addons allow! While v1 addons integrate with the app's build and as such all add a bit of overhead to it, the build step of v2 addons happens ahead of time, before they are published to npm, so at the time they are consumed by the app, they are fully static and do _not_ integrate with the app's build.
```

Comment:
the wording about not integrating with the app build at the end is a bit confusing to me. I know what you mean and I don't immediately have a different way of phrasing it, but still wanted to raise this

Reply:
I rephrased this a bit, hope makes better sense now.